molgenis_documentation/REST API.md

```diff
@@ -146,56 +157,231 @@ By default, the MOLGENIS R API retrieves its data from the server you retrieved
+```
+
+There they are.
+
+Let's format the expression counts
+
+```python
+for sample in samples:
+	print "{Ref_Counts:5}\t{Alt_Counts:5}".format(**sample)
+```
+
+```python
+130  	121
+4142 	4791
+19   	28
+19   	23
+32   	11
+639  	572
+202  	309
+423  	401
+271  	234
+806  	1081
+213  	201
+74   	96
+730  	655
+584  	699
+331  	391
+13   	14
+70   	101
+47   	35
+19   	28
+44   	47
+60   	55
+```
+
+Let's plot the expression counts in these samples in a scatter plot.
+
+```python
+import matplotlib.pyplot as plt
+plt.scatter([sample["Ref_Counts"] for sample in samples], [sample["Alt_Counts"] for sample in samples])
+plt.xlim([0, 5000])
+plt.ylim([0, 5000])
+plt.xlabel("Reference Allele")
+plt.ylabel("Alternative Allele")
+plt.title("Allele-Specific Expression for rs12460890")
+```
+
+
+And add a line for the non-specific expression.
+
+```python
+plt.plot([0, 5000], [0, 5000])
+plt.show()
+```
+![image](images/rs12460890-py.png)
 
 ## Java
+The `molgenis-data-rest-client` module contains an `org.molgenis.data.rest.clientMolgenisClient` class that we use to access the REST API for our own integration tests. It's built using the Spring `RestTemplate`.
+
+The method calls are synchronous. Not all REST API methods are supported yet, it's not well documented and it has no users that we know of other than ourselves. But if you're interested in accessing a remote MOLGENIS instance from Java, you'll want to take a look there for inspiration.
 
 ## JavaScript
 
+Internally, MOLGENIS uses `molgenis.RestClient` and `molgenis.RestClientV2` defined in `molgenis.js`. Their methods are implemented using `jquery.ajax`. This interface is not yet supported externally, but if you're planning to access MOLGENIS from JavaScript, you'll want to take a look at that for inspiration.
+
 ## HTTP
 
+This is the raw HTTP REST API that the other APIs are built upon.
+There are two versions, v1 and v2. v2 should be seen as an extension of v1.
+
+The main differences between v2 and v1 are that v2 retrieves more entity metadata and that it allows you to query using RSQL.
+But beware: the v1 csv endpoint already queries using RSQL.
 
-# Scripting
+# Scripts
 
-Scripts are defined in the Scripting plugin.
+Scripts are defined in the Scripts plugin.
 By default you can find it in the menu under Plugins, Scripts.
-The Scripting plugin allows you to edit the available scripts.
-You can also execute them by pushing the green arrows button.
+The Scripts plugin allows you to create, edit and delete the available scripts.
+
+## Creating a script
 
-The Scripting plugin also shows you a list of parameters.
+You open the script editor by clicking the add button ![image](images/new.png).
 
+### Name
```

Comment:
I wouldn't use header (#) notation for form fields. Maybe use asterisks (*) to create a list?